graphics/palette.h

```diff
@@ -153,11 +252,10 @@ class PaletteLookup {
 	 *
 	 * @return the created map, or nullptr if one isn't needed.
 	 */
-	uint32 *createMap(const byte *srcPalette, uint len, bool useNaiveAlg = false);
+	uint32 *createMap(const Palette &srcPalette, bool useNaiveAlg = false);
 
 private:
-	byte _palette[256 * 3];
-	uint _paletteSize;
+	Palette _palette;
```

Comment:
Instead of having a `Palette` object inside the `PaletteLookup`, would it be possible to make `PaletteLookup` a subclass of `Palette`?

Reply:
I think that might require the Palette::set to be virtual as changes to the palette should clear _colorHash.
We could opt to not clear it and make that a documented caveat of the lookup.

Reply:
A simpler solution might be to have a private `onChange()` virtual method that gets called by `Palette::set()` and `Palette::clear()`.

Reply:
We would also want to protect Palette's data member if going that route. I went with an accessor to the palette on the lookup for the moment.